pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -107,6 +107,7 @@ public int filterEntriesForConsumer(Optional<MessageMetadata[]> optMetadataArray
         long totalBytes = 0;
         int totalChunkedMessages = 0;
         int totalEntries = 0;
+        final boolean hasFilter = CollectionUtils.isNotEmpty(entryFilters);
         List<Position> entriesToFiltered = CollectionUtils.isNotEmpty(entryFilters) ? new ArrayList<>() : null;
```

Comment:
could we put this boolean as a class field and the replace all the `CollectionUtils.isNotEmpty(entryFilters)` ?

Reply:
yes I agree and fixed